### PR TITLE
[CMake] Conditionalize generation of .swiftinterface files

### PR DESCRIPTION
**Pull request by jrose-apple**
SWIFT_ENABLE_PARSEABLE_MODULE_INTERFACES is the new setting.

rdar://problem/44942414

### CODE REVIEW

**Comment by jrose-apple** [comment]
Still need to fix the tests that expect the swiftinterface files to be there…

**Comment by jrose-apple** [comment]
@swift-ci Please test

**Review by Rostepher — APPROVED** [review]
LGTM, the CMake changes look straightforward. 👍

File: cmake/modules/SwiftSource.cmake [hunk 1]
```diff
@@ -499,15 +508,11 @@ function(_compile_swift_files
     add_custom_command_target(
         module_dependency_target
         COMMAND
-          "${CMAKE_COMMAND}" "-E" "remove" "-f" "${module_file}"
-        COMMAND
-          "${CMAKE_COMMAND}" "-E" "remove" "-f" "${module_doc_file}"
-        COMMAND
-          "${CMAKE_COMMAND}" "-E" "remove" "-f" "${interface_file}"
+          "${CMAKE_COMMAND}" "-E" "remove" "-f" ${module_outputs}
```
Comment (compnerd) [inline cmake/modules/SwiftSource.cmake]: If this works, that is great.  I recently learnt that the multi-file variant of the commands was added in CMake 3.5 for some of them :/

File: validation-test/ParseableInterface/verify_all_overlays_O.swift [hunk 2]
```diff
@@ -1,6 +1,9 @@
...
+// generated.
+
 // RUN: %empty-directory(%t)
-// RUN: for x in %platform-sdk-overlay-dir/*.swiftinterface; do [[ $(basename "$x") = Swift.swiftinterface || $(basename "$x") = simd.swiftinterface || $(basename "$x") = SwiftLang.swiftinterface ]] && continue; %target-swift-frontend "$x" -emit-module -o %t/$(basename "$x" .swiftinterface).swiftmodule -disable-objc-attr-requires-foundation-module -enable-resilience -Fsystem %sdk/System/Library/PrivateFrameworks/ -swift-version 4 -O || echo '%target-os:' $(basename "$x") >> %t/failures.txt; done
-// RUN: diff <(grep '%target-os:' %s) <(sort -f %t/failures.txt)
+// RUN: for x in %platform-sdk-overlay-dir/*.swiftinterface; do [[ $(basename "$x") = Swift.swiftinterface || $(basename "$x") = simd.swiftinterface || $(basename "$x") = SwiftLang.swiftinterface || $(basename "$x") = '*.swiftinterface' ]] && continue; %target-swift-frontend "$x" -emit-module -o %t/$(basename "$x" .swiftinterface).swiftmodule -disable-objc-attr-requires-foundation-module -enable-resilience -Fsystem %sdk/System/Library/PrivateFrameworks/ -swift-version 4 -O || echo '%target-os:' $(basename "$x") >> %t/failures.txt; done
```
Comment (compnerd) [inline validation-test/ParseableInterface/verify_all_overlays_O.swift]: Ugh, this really makes the test difficult to port on windows :-(.

**Review by compnerd — APPROVED** [review]

File: cmake/modules/SwiftSource.cmake (hunk 1 above)
Comment (jrose-apple) [reply cmake/modules/SwiftSource.cmake]: Oh, hm. It works on my machine™ but I'll go check before I merge.

File: validation-test/ParseableInterface/verify_all_overlays_O.swift (hunk 2 above)
Comment (jrose-apple) [reply validation-test/ParseableInterface/verify_all_overlays_O.swift]: This test is kind of a mess anyway. I should probably just write a Python script. But I don't think I've made it worse of a mess than it already was.

**Comment by jrose-apple** [comment]
Unrelated XCTest failure.

@swift-ci Please test Linux

File: cmake/modules/SwiftSource.cmake (hunk 1 above)
Comment (jrose-apple) [reply cmake/modules/SwiftSource.cmake]: Looks like it was [in 3.4](https://cmake.org/cmake/help/v3.4/manual/cmake.1.html#command-line-tool-mode) at least, and our current oldest supported version is 3.4.3.

**Pull request merged**